tests/qa_automation/qaset_post_patch_run.pm

```diff
@@ -0,0 +1,148 @@
+                        </xsl:copy>
+                    </xsl:otherwise>
+                </xsl:choose>
+            </xsl:when>
+            <xsl:otherwise>
+                <xsl:copy>
+                    <xsl:apply-templates select="@*" />
+                    <xsl:apply-templates select="testcase[1]/system-err" />
+                </xsl:copy>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
+</xsl:stylesheet>
+EOT
+
+my $compare_results = <<'EOT';
+while read test id_before 
+do 
+  echo "$test,$id_before"
+  id_after=`xml sel -t -v "/testsuites/testsuite[@name=\"$test\"]/testcase[1]/system-err" /tmp/junit.xml|sed -n "s/.*id=\(.*\)/\1/p"`
+  qadb_url="http://qadb2.suse.de/qadb/regression.php?ref_submission_id=${id_before}&cand_submission_id=$id_after"
+  echo "$qadb_url"
+  xml ed -L -u "/testsuites/testsuite[@name=\"$test\"]/testcase[1]/system-err" -v "Submission comparision between before and after results: $qadb_url" /tmp/junit.xml
+done < /tmp/submission_ids_before
+EOT
+
+# qa_testset_automation validation test
+sub run {
+    my $self = shift;
+    $self->system_login();
+    $self->start_testrun();
+    my $testrun_finished = $self->wait_testrun(timeout => 180 * 60);
+
+    # Upload test logs
+    my $tarball = "/tmp/qaset.tar.bz2";
+    assert_script_run("tar cjf '$tarball' -C '/var/log/' 'qaset'");
+    upload_logs($tarball, timeout => 600);
+    my $log = $self->system_status();
+    upload_logs($log, timeout => 100);
+
+    # JUnit xml report
+    assert_script_run("/usr/share/qa/qaset/bin/junit_xml_gen.py -n 'regression' -d -o /tmp/junit.xml /var/log/qaset");
+
+    # Record into failure for known bugs to aviod triggering whole test failure.
+    if (my %softfail_tc = @{get_var_array('SOFTFAIL_TESTCASES')}) {
+        my $broken_tc_list;
+        while (my ($k, $v) = each %softfail_tc) {
+            next unless script_run("grep $k /tmp/broken_tclist");
+            $broken_tc_list .= "$k ";
+            record_info("$k => $v");
+        }
+        assert_script_run(qq(echo "`cat /tmp/broken_tclist` $broken_tc_list" > /tmp/broken_tclist ));
+    }
+
+    upload_logs("/tmp/junit.xml", timeout => 600);
+    # Construct qadb regression url
+    assert_script_run("cat > /tmp/compare_results.sh <<'END'\n$compare_results\nEND\n( exit \$?)");
+    assert_script_run("bash /tmp/compare_results.sh");
+
+    # transform junit.xml by xslt to filter the result of passed, broken and skipped testcases. Only new regression is kept;
+    assert_script_run("cat > /tmp/trans_junit.xsl <<'END'\n$xslt\nEND\n( exit \$?)");
+    assert_script_run("xsltproc  --stringparam var \"`cat /tmp/broken_tclist`\" /tmp/trans_junit.xsl /tmp/junit.xml > /tmp/junit_trans.xml");
+
+
+    # upload to openqa
+    parse_junit_log("/tmp/junit_trans.xml");
+
+    unless ($testrun_finished) {
+        die "Test run didn't finish within time limit";
+    }
+
+    # Switch back to root-console
```

Comment:
I guess this comment is obvious and not necessary :) Same in line 137

Reply:
Deleted it.